0.20.7 blog post (#211)
TeamNewPipe/website: _posts/2020-12-24-newpipe-0.20.7.md

---
layout: post
title: "NewPipe 0.20.7 released: YouTube Mix support, bug fixes and many improvements"
short: "NewPipe 0.20.7 released"
date: 2020-12-24 19:00:00
categories: [pinned, release]
author: Poolitzer
image: newpipe
excerpt_separator: <!-- more -->
---

Merry Christmas, if you celebrate that, and almost Happy New Year, if you use a calendar at least close to the Gregorian one. We want to present you a little gift as a good way to start into 2021 AD: The next release of NewPipe. It comes with a lot of improvements, fixes, and even a big new feature. So let us dive into this release post!

<!-- more -->

## New in this version
YouTube mixes are a feature now, and they are dropping in hot (and with a lot of bass). Opening a mix in previous versions resulted in the first video of the mix playing, nothing else. Now you see the full playlist and even a nice radio icon as an indicator. Enjoy listening to never-ending lists and thank [@XiangRongLin](https://github.com/XiangRongLin) for this feature as well as [@Stypox](https://github.com/Stypox) for reviewing it.

> This release includes support for another feature from YouTube: meta info. The little boxes sometimes telling you that a channel belongs to a news group, or giving you further information to specific search queries - they appear in NewPipe now as well. If you do not want to see those boxes, head over to <code><nobr><img src="{{ site.baseurl }}/img/icons/baseline-settings-20px.svg" /> Settings</nobr> > <nobr><img src="{{ site.baseurl }}/img/icons/baseline-language-20px.svg" /> Content</nobr> > <nobr>Show meta info</nobr></code> and disable the display there. If you enjoy these boxes however, thank [@TobiGr](https://github.com/TobiGr) for bringing them into this release.

Reviewer: ```suggestion
This release includes support for another feature from YouTube: meta info. The little boxes sometimes telling you that a channel belongs to a public broadcaster, or giving you further information to specific search queries - they appear in NewPipe now as well. If you do not want to see those boxes, head over to <code><nobr><img src="{{ site.baseurl }}/img/icons/baseline-settings-20px.svg" /> Settings</nobr> > <nobr><img src="{{ site.baseurl }}/img/icons/baseline-language-20px.svg" /> Content</nobr> > <nobr>Show meta info</nobr></code> and disable the display there. If you enjoy these boxes however, thank [@TobiGr](https://github.com/TobiGr) for bringing them into this release.
```

---
layout: post
title: "NewPipe 0.20.7 released: YouTube Mix support, bug fixes and many improvements"
short: "NewPipe 0.20.7 released"
date: 2020-12-24 19:00:00
categories: [pinned, release]
author: Poolitzer
image: newpipe
excerpt_separator: <!-- more -->
---

> Merry christmas, if you celebrate that, and almost happy new year, if you use a calender at least close to the gregorian one. We want to present you a little gift as a good way to start into 2021 AD: The next release of NewPipe. It comes with a lot of improvement, fixes, and even a big new feature. So let us dive into this release post

Reviewer 2: ```suggestion
Merry Christmas, if you celebrate that, and almost happy new year, if you use a calendar at least close to the Gregorian one. We want to present you a little gift as a good way to start into 2021 AD: The next release of NewPipe. It comes with a lot of improvements, fixes, and even a big new feature. So let us dive into this release post!
```

Author: looks good, we need to check if we make this a new years release or in between the days and change the first sentence accordingly

<!-- more -->

## New in this version

> YouTube mixes are a feature now, and they are dropping in hot (and with a lot of bass). Opening a mix in previous versions resulted in the first video of the mix playing, nothing else. Now you see the full playlist and even a nice radio icon as an indicator. Enjoy listing to never ending lists and thank [@Stypox](Stypox) as well as [@XiangRongLin](https://github.com/XiangRongLin) for this feature.

Reviewer 2: ```suggestion
YouTube mixes are a feature now, and they are dropping in hot (and with a lot of bass). Opening a mix in previous versions resulted in the first video of the mix playing, nothing else. Now you see the full playlist and even a nice radio icon as an indicator. Enjoy listening to never ending lists and thank [@Stypox](Stypox) as well as [@XiangRongLin](https://github.com/XiangRongLin) for this feature.
```
Also, does `[@Stypox](Stypox)` actually work?

Author: No it does not my regex and me had a disagreement it seems.

> YouTube mixes are a feature now, and they are dropping in hot (and with a lot of bass). Opening a mix in previous versions resulted in the first video of the mix playing, nothing else. Now you see the full playlist and even a nice radio icon as an indicator. Enjoy listening to never-ending lists and thank [@Stypox](https://github.com/Stypox) as well as [@XiangRongLin](https://github.com/XiangRongLin) for this feature.

Reviewer 3: I only reviewed, I'd give full credit to @XiangRongLin ;-)
If it's too late to suggest then nevermind
```suggestion
YouTube mixes are a feature now, and they are dropping in hot (and with a lot of bass). Opening a mix in previous versions resulted in the first video of the mix playing, nothing else. Now you see the full playlist and even a nice radio icon as an indicator. Enjoy listening to never-ending lists and thank [@XiangRongLin](https://github.com/XiangRongLin) for this feature.
```

Author: no thats fine but Im gonna still include you ;P cant escape.

```suggestion
YouTube mixes are a feature now, and they are dropping in hot (and with a lot of bass). Opening a mix in previous versions resulted in the first video of the mix playing, nothing else. Now you see the full playlist and even a nice radio icon as an indicator. Enjoy listening to never-ending lists and thank [@XiangRongLin](https://github.com/XiangRongLin) for this feature as well as [@Stypox](https://github.com/Stypox) for reviewing it.
```

how do you like this?

Reviewer 3: Ok, fine :-D

Author: good.

> This release includes support for another feature from YouTube: video metadata. The little boxes that appear sometimes - telling you that a channel belongs to a public broadcaster, or giving you further information regarding specific search queries - appear in NewPipe as well now. If you do not want to see them, head over to <code><nobr><img src="{{ site.baseurl }}/img/icons/baseline-settings-20px.svg" /> Settings</nobr> > <nobr><img src="{{ site.baseurl }}/img/icons/baseline-language-20px.svg" /> Content</nobr> > <nobr>Show meta info</nobr></code> and disable the toggle there. If you enjoy these boxes however, thank [@TobiGr](https://github.com/TobiGr) for baking them into this release.

Author: ```suggestion
This release includes support for another feature from YouTube: video metadata. The little boxes that appear sometimes - telling you that a channel belongs to a public broadcaster, or giving you further information regarding specific search queries - appear in NewPipe as well now. If you do not want to see them, head over to <code><nobr><img src="{{ site.baseurl }}/img/icons/baseline-settings-20px.svg" /> Settings</nobr> > <nobr><img src="{{ site.baseurl }}/img/icons/baseline-language-20px.svg" /> Content</nobr> > <nobr>Show meta info</nobr></code> and disable the toggle there. If you enjoy these boxes however, thank [@TobiGr](https://github.com/TobiGr) for bringing them into this release.
```

more clear for non coders

Reviewer 4: Baking is not a coding term. xD

But it's fine with me either way.

Reviewer: I am fine with both. If you prefer bringing, we can use that.

In some kind of more personal news, we have a new domain now - [https://newpipe.net](https://newpipe.net). This makes us look more professional (we hope), but also meant the URLs needed to be updated in the app. This was done by [@TiA4f8R](https://github.com/TiA4f8R).
## Bugs, be gone!

> We aim for the best NewPipe with every release we do. Because of that, we improve on issues in every update. This time around, we fixed an issue where the app would crash when you tried to delete a video from your playlist while the app was refreshing your subscriptions. This was discovered by [@gasull](gasull), reproduced by [@chilliger](https://github.com/chilliger) and fixed by [@hlloreda](https://github.com/hlloreda). Thanks to all three of them. 

Reviewer 2: Same above question for `[@gasull](gasull)`.

Author: ```suggestion
We aim for the best NewPipe with every release we do. Because of that, we improve on issues in every update. This time around, we fixed an issue where the app would crash when you tried to delete a video from your playlist while the app was refreshing your subscriptions. This was discovered by [@gasull](https://github.com/gasull), reproduced by [@chilliger](https://github.com/chilliger) and fixed by [@hlloreda](https://github.com/hlloreda). Thanks to all three of them. 
```

In another attempt to improve the ReCaptcha experience, [@Stypox](https://github.com/Stypox) fixed a situation where only a blank page was shown and is likely very exhausted with that topic at this point. We thank him massively for taking up this issue time and time again.

> Talking about time, sharing URLs from any service except YouTube now doesn't try to add a timestamp to the URL of the video. This broke the URL on any other service. In the future we hope to support the time stamps of every platform, for now, you get non timed URLs that work. This was provided by [@B0pol](https://github.com/B0pol), thank you!

Reviewer 2: ```suggestion
Talking about time, sharing URLs from any service except YouTube now doesn't try to add a timestamp to the URL of the video. This broke the URL on any other service. In the future we hope to support the timestamps of every platform, for now, you get non-timed URLs that work. This was provided by [@B0pol](https://github.com/B0pol), thank you!
```
From my quick search, it seems that both "non-timed" and "nontimed" work. Choose whichever one you want.

Author: more - is all I want (for christmas)

> The last big fix is brought to you by [@TobiGr](https://github.com/TobiGr), who got YouTube's music search to work once again. Search all the music that you want now (with the filters next to the search bar) and enjoy listing to christmas songs you can not get out of your head.

Reviewer 2: ```suggestion
The last big fix is brought to you by [@TobiGr](https://github.com/TobiGr), who got YouTube's music search to work once again. Search all the music that you want now (with the filters next to the search bar) and enjoy listing to Christmas songs you can not get out of your head.
```

> Talking about time - when sharing video URLs from services other than YouTube, the app doesn't try to append a timestamp to the URL anymore, which used to make them invalid. In the future, we hope to support the timestamp formats of every platform, but for now, you get non-timestamped URLs that work. This was provided by [@B0pol](https://github.com/B0pol). Thank you!

Author: ```suggestion
Talking about time - when sharing video URLs from services other than YouTube, the app doesn't try to append a timestamp to the URL anymore, which makes them invalid. In the future, we hope to support the timestamp formats of every platform, but for now, you get non-timestamped URLs that work. This was provided by [@B0pol](https://github.com/B0pol). Thank you!
```

Reviewer 4: Now it sounds like the new behaviour makes them invalid.

Reviewer: I'd also keep the current version.

## Nerd talk

> There were quite some very technical/non user facing changes in this release. We decided to provide you this nice list with changes not affecting you at all, but we still want to publicly thank their creators:

Reviewer 2: ```suggestion
There were quite some very technical/non-user-facing changes in this release. We decided to provide you this nice list with changes not affecting you at all, but we still want to publicly thank their creators:
```

> The last big fix is brought to you by [@TobiGr](https://github.com/TobiGr), who got YouTube Music's search to work once again. Search all the music that you want now (with the filters in the 3-dot menu next to the search bar) and enjoy listing to Christmas songs that you cannot get out of your head.

Author: ```suggestion
The last big fix is brought to you by [@TobiGr](https://github.com/TobiGr), who got YouTube Music's search to work once again. Search all the music that you want now (with the filters in the 3-dot menu next to the search bar) and enjoy listing to Christmas songs which you cannot get out of your head.
```

Reviewer 4: Better.

- The Invidious (alternative YouTube frontend which the extractor uses to access YouTube) URL list in the manifest was updated by [@opusforlife2](https://github.com/opusforlife2)

> - The ReCaptcha user agent doesn't include device informations anymore, by [@Stypox](https://github.com/Stypox) (to no ones suprise)

Reviewer 3: This could potentially be of interest and impact someone, do you think it would be a bad idea to put it above, together with the other note about recaptchas?

Author: hmm I thought about it but also thought it would be too technical. I definitly can though. How does that sound to you?

In another attempt to improve the ReCaptcha experience, [@Stypox](https://github.com/Stypox) fixed a situation where only a blank page was shown. He also changed that the user agent sent to the ReCaptcha service does not include device information anymore. He is likely very exhausted with that topic at this point. We thank him massively for taking up this issue time and time again.

Reviewer 3: What about "He also prevented device information to be leaked inside the user agent"

Author: sure if you prefere leaked.

short: "NewPipe 0.20.7 released"
date: 2020-12-24 19:00:00
categories: [pinned, release]
author: Poolitzer, Opusforlife
image: newpipe
excerpt_separator: <!-- more -->
---

Merry Christmas, if you celebrate that, and almost Happy New Year, if you use a calendar at least close to the Gregorian one. We want to present you a little gift as a good way to start 2021 AD/CE (or the year 12021 in the [Human Era](https://en.wikipedia.org/wiki/Holocene_calendar), according to Kurzgesagt): The next release of NewPipe. It comes with a lot of improvements, fixes, and even a big new feature. So let us dive into this release post!

<!-- more -->

## New in this version
YouTube Mixes are a thing now, and they are dropping in hot (and with a lot of bass). Opening a mix in previous versions resulted in just the first video of the mix playing, nothing else. Now you see the full playlist and even a nice radio icon as an indicator. Enjoy listening to never-ending playlists and thank [@XiangRongLin](https://github.com/XiangRongLin) for this feature, as well as [@Stypox](https://github.com/Stypox) for reviewing it thoroughly.

> This release includes support for another feature from YouTube: video metadata. The little boxes that appear sometimes - telling you that a channel belongs to a news group, or giving you further information regarding specific search queries - appear in NewPipe as well now. If you do not want to see them, head over to <code><nobr><img src="{{ site.baseurl }}/img/icons/baseline-settings-20px.svg" /> Settings</nobr> > <nobr><img src="{{ site.baseurl }}/img/icons/baseline-language-20px.svg" /> Content</nobr> > <nobr>Show meta info</nobr></code> and disable the toggle there. If you enjoy these boxes however, thank [@TobiGr](https://github.com/TobiGr) for baking them into this release.

Reviewer: ```suggestion
This release includes support for another feature from YouTube: video metadata. The little boxes that appear sometimes - telling you that a channel belongs to a public broadcaster, or giving you further information regarding specific search queries - appear in NewPipe as well now. If you do not want to see them, head over to <code><nobr><img src="{{ site.baseurl }}/img/icons/baseline-settings-20px.svg" /> Settings</nobr> > <nobr><img src="{{ site.baseurl }}/img/icons/baseline-language-20px.svg" /> Content</nobr> > <nobr>Show meta info</nobr></code> and disable the toggle there. If you enjoy these boxes however, thank [@TobiGr](https://github.com/TobiGr) for baking them into this release.
```

- Our CI tests were switched from Travis to GitHub by [@FireMasterK](https://github.com/FireMasterK)
- Some internal variables and comments had typos, and [@mhmdanas](https://github.com/mhmdanas) fixed them
- We use `ContextCompat.getDataDir()`, `ServiceCompat.stopForeground()` and `ActivityCompat.recreate()` now, because [@Isira-Seneviratne](https://github.com/Isira-Seneviratne) decided that is good and everyone agreed, and also because older Android versions can now use the existing code without needing special caveats and corner case handling
- [@Isira-Seneviratne](https://github.com/Isira-Seneviratne) changed how the file hash calculation in the Downloads menu is displayed - it does not use ProgressDialog anymore because that is deprecated
- The database export logic got its own class so that we can create unit tests for it, thanks to [@XiangRongLin](https://github.com/XiangRongLin)
- [@Isira-Seneviratne](https://github.com/Isira-Seneviratne) is on fire this release; they also changed Kotlin synthetics to view bindings because of deprecation of the former

> - The NewPipe extractor supports stream segments for YouTube now, which has been included in this release, because of [@vkay94](https://github.com/vkay94)

Reviewer: ```suggestion
- The NewPipe extractor supports stream segments for YouTube now, which has been included in this release, because of [@vkay94](https://github.com/vkay94). [@vkay94](https://github.com/vkay94) is currently working on integrating the changes into the app.
```

Reviewer 4: Oh. If it's not there in the front-end, we should just remove it altogether. When the UI PR is merged, this can be added in one go.

Reviewer: I agree
